main_test.go

```diff
@@ -96,13 +96,14 @@ func TestGenerate(t *testing.T) {
 		{
 			desc: "happy case with v3 config by default",
 			input: configInput{
-				xdsServerUri:      "example.com:443",
-				gcpProjectNumber:  123456789012345,
-				vpcNetworkName:    "thedefault",
-				ip:                "10.9.8.7",
-				zone:              "uscentral-5",
-				metadataLabels:    map[string]string{"k1": "v1", "k2": "v2"},
-				includeV3Features: true,
+				xdsServerUri:         "example.com:443",
+				gcpProjectNumber:     123456789012345,
+				vpcNetworkName:       "thedefault",
+				ip:                   "10.9.8.7",
+				zone:                 "uscentral-5",
+				metadataLabels:       map[string]string{"k1": "v1", "k2": "v2"},
+				includeV3Features:    true,
+				includeXDSFederation: true,
```

Comment:
Avoid changing this test. Instead add a new one like how each of the features has its own.

Reply:
sg

Reply:
my bad actually. adding one more v3 and v2 config cases